src/server/svr_jobfunc.c

```diff
@@ -5408,6 +5408,16 @@ svr_histjob_update(job * pjob, int newstate, int newsubstate)
 	pjob->ji_qs.ji_substate = newsubstate;
 	set_statechar(pjob);
 
+	/*
+	 * If subjob, set the substate and purge the job structure
+	 * right now and return.
+	 */
+	if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob) {
+		if (JOB_STATE_FINISHED == newstate)
+			newstate = JOB_STATE_EXPIRED;
```

Comment:
same as the above comment.

Reply:
Since subjob never go to F state (instead go to X) , the if block is added to guard if any one setting it to F state.

Reply:
Can you update the same in comments. That helps people reading the code why it is done.

Reply:
Added in rc 13 https://github.com/PBSPro/pbspro/pull/590/commits/82b8f337b566ec556777394e4e3ef00fc07a0d4b